### PR TITLE
Add mandatory SLULA license acceptance flag (close #405)

### LINKED ISSUE #405: Add mandatory SLULA license acceptance flag
Since introducing a new license we need to explicitly check if the user has accepted the terms. Therefore a new flag is added.

### PR DESCRIPTION
**Pull request by peel**
Since introducing a new license we need to explicitly check if the user has accepted the terms. Therefore a new flag is added. By default it is set to false but can be overrided by either:
- setting `license.accept = { true | yes | on }` in the config file
- setting `env ACCEPT_LIMITED_USE_LICENSE={ true | yes | on }`
- appending `-Dlicense.accept={ true | yes | on }`

Failing to accept the license will shut down the app and report a RuntimeException:
```
java.lang.RuntimeException: License not accepted. Configure license acceptance to continue. See docs for more details.
```

### CODE REVIEW

File: core/src/main/resources/reference.conf [hunk 1]
```diff
@@ -1,4 +1,9 @@
 {
+  license {
+    accept = false
+    accept = ${?ACCEPT_LIMITED_USE_LICENSE}
```
[the remark this answers is not in the document]
Comment (peel) [reply core/src/main/resources/reference.conf:4]: Yes, that's exactly what the behaviour is.

File: core/src/main/scala/com.snowplowanalytics.snowplow.collector.core/Run.scala [hunk 2]
```diff
@@ -54,15 +54,24 @@ object Run {
...
-      Logger[F].error(e)("Exiting") >>
-        prettyLogException(e).as(ExitCode.Error)
-    }
+    eitherT
+      .merge
+      .ensure(
+        new RuntimeException(
+          "License not accepted. Configure license acceptance to continue. See docs for more details."
+        )
+      )(_ != licenseNotAccepted)
```
[the remark this answers is not in the document]
Comment (peel) [reply core/src/main/scala/com.snowplowanalytics.snowplow.collector.core/Run.scala]: Yea, agreed. It's much better. I've rushed my implementation and did not really like it as well.

File: core/src/main/scala/com.snowplowanalytics.snowplow.collector.core/Run.scala [hunk 3]
```diff
@@ -54,15 +54,24 @@ object Run {
...
 
-    eitherT.merge.handleErrorWith { e =>
-      Logger[F].error(e)("Exiting") >>
-        prettyLogException(e).as(ExitCode.Error)
-    }
+    eitherT
+      .merge
+      .ensure(
+        new RuntimeException(
+          "License not accepted. Configure license acceptance to continue. See docs for more details."
```
[the remark this answers is not in the document]
Comment (peel) [reply core/src/main/scala/com.snowplowanalytics.snowplow.collector.core/Run.scala]: That was my intention, but maybe @stanch thinks differently?

File: core/src/main/scala/com.snowplowanalytics.snowplow.collector.core/Config.scala [hunk 4]
```diff
@@ -152,7 +153,17 @@ object Config {
...
+    accept: Boolean
+  )
+
   implicit def decoder[SinkConfig: Decoder]: Decoder[Config[SinkConfig]] = {
+    implicit val license: Decoder[License] = {
+      val truthy = Set("true", "yes", "on", "1")
+      Decoder
+        .forProduct1("accept")((s: String) => License(truthy(s.toLowerCase())))
+        .or(Decoder.forProduct1("accept")((b: Boolean) => License(b)))
+    }
```
Comment (benjben) [inline core/src/main/scala/com.snowplowanalytics.snowplow.collector.core/Config.scala:166]: Much more concise :+1:

**Review by benjben — APPROVED** [review]
Looks great !

**Comment by peel** [comment]
Pulled into `release/3.0.0`

**Pull request closed without merging**